### PR TITLE
develop: refactor SQLAlchemy dev guide

### PR DESCRIPTION
**Pull request by wd0517**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 1]
```diff
@@ -1,249 +1,293 @@
...
-def trade_check(session: Session, sell_id: str, buy_id: str, amount: int, price: int) -> bool:
-    # sell player goods check
-    sell_player = session.query(Player.goods).filter(Player.id == sell_id).with_for_update().one()
-    if sell_player.goods < amount:
-        print(f'sell player {sell_id} goods not enough')
-        return False
+    ```dotenv
+    TIDB_HOST='{gateway-region}.aws.tidbcloud.com'
+    TIDB_PORT='4000'
+    TIDB_USER='{prefix}.root'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:110]:
```suggestion
    TIDB_USER='{user}'
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 2]
```diff
@@ -1,249 +1,293 @@
...
+6. Copy and paste the corresponding connection string into the `.env` file. The example result is as follows:
 
-def trade_check(session: Session, sell_id: str, buy_id: str, amount: int, price: int) -> bool:
-    # sell player goods check
-    sell_player = session.query(Player.goods).filter(Player.id == sell_id).with_for_update().one()
-    if sell_player.goods < amount:
-        print(f'sell player {sell_id} goods not enough')
-        return False
+    ```dotenv
+    TIDB_HOST='{gateway-region}.aws.tidbcloud.com'
```
Comment (sykp241095) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:108]:
```suggestion
    TIDB_HOST='{host}'
```

There is no concept of "gateway-region" in tidbcloud.

**Review by Mini256 — APPROVED** [review]

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 3]
```diff
@@ -1,249 +1,294 @@
...
-from sqlalchemy.orm import declarative_base, sessionmaker
+```shell
+pip install -r requirements.txt
+```
 
-engine = create_engine('mysql://root:@127.0.0.1:4000/test')
-Base = declarative_base()
-Base.metadata.create_all(engine)
-Session = sessionmaker(bind=engine)
+#### Why do we need PyMySQL?
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:66]:
```suggestion
#### Why use PyMySQL?
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 4]
```diff
@@ -1,249 +1,294 @@
...
 
+SQLAlchemy is an ORM library that supports multiple databases. It is a high-level abstraction of the database, which can help us write SQL statements in a more object-oriented way. However, it does not provide a database driver. We need to install a database driver to connect to the database. In this sample project, we use PyMySQL as the database driver, which is a pure Python MySQL client library that is compatible with TiDB, can be easily installed in all platforms.
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:68]:
```suggestion
SQLAlchemy is an ORM library that works with multiple databases. It provides a high-level abstraction of the database, which helps developers write SQL statements in a more object-oriented way. However, SQLAlchemy does not include a database driver. To connect to a database, you need to install a database driver. This sample application uses PyMySQL as the database driver, which is a pure Python MySQL client library that is compatible with TiDB and can be installed on all platforms.
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 5]
```diff
@@ -1,249 +1,294 @@
...
 
-class Player(Base):
-    __tablename__ = "player"
+You can also use other database drivers, such as [mysqlclient](https://github.com/PyMySQL/mysqlclient) and [mysql-connector-python](https://dev.mysql.com/doc/connector-python/en/), but they are not pure Python libraries, so you need to install the corresponding C/C++ compiler and MySQL client library to compile them. For more information, refer to [SQLAlchemy's official documentation](https://docs.sqlalchemy.org/en/20/core/engines.html#mysql).
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:70]:
```suggestion
You can also use other database drivers, such as [mysqlclient](https://github.com/PyMySQL/mysqlclient) and [mysql-connector-python](https://dev.mysql.com/doc/connector-python/en/). But they are not pure Python libraries and require the corresponding C/C++ compiler and MySQL client library to compile them. For more information, refer to [SQLAlchemy official documentation](https://docs.sqlalchemy.org/en/20/core/engines.html#mysql).
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 6]
```diff
@@ -1,249 +1,294 @@
...
-</CustomContent>
+### Step 1: Clone the sample app repository
 
-## Step 2. Get the code
+Run the following commands in your terminal window to clone the sample code repository:
 
 ```shell
-git clone https://github.com/pingcap-inc/tidb-example-python.git
+git clone https://github.com/tidb-samples/tidb-python-sqlalchemy-quickstart.git
+cd tidb-python-sqlalchemy-quickstart;
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:55]:
```suggestion
cd tidb-python-sqlalchemy-quickstart
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 7]
```diff
@@ -1,249 +1,294 @@
...
 ```
 
-The following uses SQLAlchemy 1.44 as an example.
+### Step 2: Install dependencies
 
-```python
-import uuid
-from typing import List
+Run the following command to install the required packages (including `SQLAlchemy` and `PyMySQL`) for the sample app:
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:60]:
```suggestion
Run the following command to install the required packages (including SQLAlchemy and PyMySQL) for the sample app:
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 8]
```diff
@@ -1,249 +1,294 @@
...
+    session.delete(player)
+    session.commit()
 ```
 
-## Step 4. Expected output
+For more information, refer to [Delete data](/develop/dev-guide-delete-data.md).
+
+## Next steps
+
+- Learn more usage of `SQLAlchemy` from [the documentation of SQLAlchemy](https://www.sqlalchemy.org/).
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:288]:
```suggestion
- Learn more usage of SQLAlchemy from [the documentation of SQLAlchemy](https://www.sqlalchemy.org/).
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 9]
```diff
@@ -1,249 +1,296 @@
 ---
-title: Build a Simple CRUD App with TiDB and SQLAlchemy
-summary: Learn how to build a simple CRUD application with TiDB and SQLAlchemy.
-aliases: ['/tidb/dev/dev-guide-outdated-for-sqlalchemy']
+title: Connect to TiDB with SQLAlchemy
+summary: Learn how to connect to TiDB using SQLAlchemy. This tutorial gives Python sample code snippets that work with TiDB using SQLAlchemy.
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:3]:
```suggestion
summary: Learn how to connect to TiDB using SQLAlchemy. This tutorial gives Python sample code snippets that work with TiDB using SQLAlchemy.
aliases: ['/tidb/dev/dev-guide-outdated-for-sqlalchemy']
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 10]
```diff
@@ -1,249 +1,296 @@
...
-Base = declarative_base()
-Base.metadata.create_all(engine)
-Session = sessionmaker(bind=engine)
+#### Why use PyMySQL?
 
+SQLAlchemy is an ORM library that works with multiple databases. It provides a high-level abstraction of the database, which helps developers write SQL statements in a more object-oriented way. However, SQLAlchemy does not include a database driver. To connect to a database, you need to install a database driver. This sample application uses PyMySQL as the database driver, which is a pure Python MySQL client library that is compatible with TiDB and can be installed on all platforms.
 
-class Player(Base):
-    __tablename__ = "player"
+You can also use other database drivers, such as [mysqlclient](https://github.com/PyMySQL/mysqlclient) and [mysql-connector-python](https://dev.mysql.com/doc/connector-python/en/). But they are not pure Python libraries and require the corresponding C/C++ compiler and MySQL client library to compile them. For more information, refer to [SQLAlchemy official documentation](https://docs.sqlalchemy.org/en/20/core/engines.html#mysql).
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:70]:
```suggestion
You can also use other database drivers, such as [mysqlclient](https://github.com/PyMySQL/mysqlclient) and [mysql-connector-python](https://dev.mysql.com/doc/connector-python/en/). But they are not pure Python libraries and require the corresponding C/C++ compiler and MySQL client for compiling. For more information, refer to [SQLAlchemy official documentation](https://docs.sqlalchemy.org/en/20/core/engines.html#mysql).
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 11]
```diff
@@ -1,249 +1,296 @@
...
+    id = Column(Integer, primary_key=True)
+    name = Column(String(32), unique=True)
+    coins = Column(Integer)
+    goods = Column(Integer)
+
+    __tablename__ = "players"
 ```
 
-Suppose that the password you set is `123456`, and the connection parameters you get from the cluster details page are the following:
+For more information, refer to [SQLAlchemy Mapping Classes with Declarative](https://docs.sqlalchemy.org/en/20/orm/declarative_mapping.html).
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:243]:
```suggestion
For more information, refer to [SQLAlchemy documentation: Mapping classes with declarative](https://docs.sqlalchemy.org/en/20/orm/declarative_mapping.html).
```

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**